### PR TITLE
docs: use titles for more code blocks

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Use titles for more code blocks
- Other small style fixes

## Context

Change code blocks so they show the file name or description of the content.

Some introductory text for code blocks is too long or complex to put in the `title=""`, so I did not change those. 😄

Read the [Material for MkDocs, adding a title to code blocks](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title) to learn more, and to see an example.

Follow-up PR after:

- #25496

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/examples/self-hosting.md [hunk 1]
```diff
@@ -277,7 +274,7 @@ renovate
 ```
 
 Save the script file, and run the script manually.
-Only add the script to cron after you checked it works.
+Only add the script to `cron`` after you checked it works.
```
Comment (HonkingGoose) [inline docs/usage/examples/self-hosting.md:277]:
```suggestion
Only add the script to `cron` after you checked it works.
```

File: docs/usage/key-concepts/automerge.md [hunk 2]
```diff
@@ -52,7 +51,7 @@ Automerging lint tool updates can be a real time-saver.
 Sometimes an update to a lint tool or plugin definition causes tests to fail, and that is usually deliberate/intentional because the lint authors have added a new rule that you need to adhere to.
 But in many cases the new version(s) will pass tests, and if so then there's really nothing else to consider before merging, so they may as well be automerged:
 
-```json
+```json title="Example of automerging lint and prettier development packages"
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/automerge.md:54]:
```suggestion
```json title="Example of automerging lint and Prettier development packages"
```

**Comment by HonkingGoose** [comment]
> How do these look when viewed in "preview" form on GitHub?

GitHub does _not_ show the title in the codeblock. So the published docs look better, but the "nice GitHub preview of the Markdown file" gets broken. When you view the `raw` file or go into the edit mode, you'll see the titles of course.

**Comment by rarkins** [comment]
@viceice @secustor wdyt?

**Comment by secustor** [comment]
As long we only do it for user docs the broken preview is fine for me.

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**